File: backend/app/models/tile.rb
class Tile
  BLUE = "blue"
  RED = "red"
  BLACK = "black"
  LIGHT_BLUE = "light_blue"
  YELLOW = "yellow"
  FIRST = "first"

  COLORS = [
    BLUE,
    RED,
    BLACK,
    LIGHT_BLUE,
    YELLOW
  ]

  attr_accessor :id, :color, :game_id

  def initialize(args = {})
    args = args&.with_indifferent_access
    @id = args.fetch(:id, nil)
    @game_id = args.fetch(:game_id, nil)
    @color = args.fetch(:color, nil)
  end

  def attributes
    {
      id: id,
      game_id: game_id,
      color: color
    }.compact
  end
end

File: frontend/src/components/tile/tile.js
import React from 'react';
import styles from './tile.module.css';

function Tile(props) {
  let stylesClasses = [styles.Tile]

  switch (props.color) {
    case 'first':
      stylesClasses.push(styles.First);
      break;
    case 'yellow':
      stylesClasses.push(styles.Yellow);
      break;
    case 'red':
      stylesClasses.push(styles.Red);
      break;
    case 'blue':
      stylesClasses.push(styles.Blue);
      break;
    case 'light_blue':
      stylesClasses.push(styles.LightBlue);
      break;
    case 'black':
      stylesClasses.push(styles.Black);
      break;
    default:
      stylesClasses.push(styles.Empty);
      break;
  }

  if (props.inactive) {
    stylesClasses.push(styles.Inactive);
  }

  if (props.selected) {
    stylesClasses.push(styles.Selected);
  }

  return (
    <div className={stylesClasses.join(' ')}>
    </div>
  );
}

export default Tile;
